plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotSegmentPageSource.java

```diff
@@ -342,15 +343,28 @@ private Block getArrayBlock(int rowIndex, int columnIndex)
     private Slice getSlice(int rowIndex, int columnIndex)
     {
         Type trinoType = getType(columnIndex);
+        DataTable dataTable = currentDataTable.getDataTable();
+
         if (trinoType instanceof VarcharType) {
-            String field = currentDataTable.getDataTable().getString(rowIndex, columnIndex);
+            String field = dataTable.getString(rowIndex, columnIndex);
             return getUtf8Slice(field);
         }
         if (trinoType instanceof VarbinaryType) {
-            return Slices.wrappedBuffer(toBytes(currentDataTable.getDataTable().getString(rowIndex, columnIndex)));
+            if (dataTable.getVersion() >= 4) {
+                try {
+                    return Slices.wrappedBuffer(dataTable.getBytes(rowIndex, columnIndex).getBytes());
+                }
+                catch (NullPointerException e) {
+                    // Pinot throws NPE when the entry is null.
```

Comment:
This behavior is weird to me. There's `org.apache.pinot.common.datatable.DataTable#getNullRowIds` method. Is it possible to use the method instead?

Reply:
No, apparently there is an issue in Pinot DataTableV4 for Pinot 0.12.1 to read null value(I need to verify if this is fixed in 1.0), this exception is captured by the test `testNullBehavior`.

Reply:
It would be nice to file an issue in Pinot repository.

Reply:
Yes. I'm tracking this.

Reply:
This is fixed in pinot 1.0.0 release: https://github.com/apache/pinot/pull/10543/files#diff-2276298496d8c930553150ded2ebe21d7caa2b809e450a98618d1bd655c04891L152